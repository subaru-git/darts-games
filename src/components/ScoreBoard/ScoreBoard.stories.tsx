import type { Meta, StoryObj } from '@storybook/react';
import ScoreBoard from './ScoreBoard';

const meta: Meta<typeof ScoreBoard> = { component: ScoreBoard, title: 'Components/ScoreBoard' };

export default meta;
type Story = StoryObj<typeof ScoreBoard>;
export const Default: Story = {
  args: {
    data: [
      {
        Scored: '',
        ToGo: '501',
        Hits: null,
      },
      {
        Scored: '180',
        ToGo: '321',
        Hits: ['T20', 'T20', 'T20'],
      },
      {
        Scored: '180',
        ToGo: '141',
        Hits: ['T20', 'T20', 'T20'],
      },
      {
        Scored: '129',
        ToGo: '12',
        Hits: ['T20', 'T19', '12'],
      },
      {
        Scored: 'B',
        ToGo: '12',
        Hits: ['D12', '0', '0'],
      },
      {
        Scored: '-',
        ToGo: '12',
        Hits: ['OUT', 'OUT', 'OUT'],
      },
      {
        Scored: '',
        ToGo: 'â‘¡',
        Hits: ['OUT', 'D6', '0'],
      },
    ],
  },
};
