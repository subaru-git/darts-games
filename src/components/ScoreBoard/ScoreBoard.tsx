import React, { FC } from 'react';

type ScoreBoardProps = {
  data: GameScore[];
};

const ScoreBoard: FC<ScoreBoardProps> = ({ data }) => {
  return (
    <table className='table-fixed border-collapse'>
      <thead>
        <tr>
          <th className='border-collapse border border-gray-400 px-6 py-3'>Scored</th>
          <th className='border-collapse border border-gray-400 px-6 py-3'>To Go</th>
          <th className='border-collapse border border-gray-400 px-6 py-3'>Darts</th>
          <th className='border-collapse border border-gray-400 px-6 py-3'>Hits</th>
        </tr>
      </thead>
      <tbody>
        {data.map((d, i) => {
          return (
            <tr key={`${d.Scored}-${i}`}>
              <td className='border-collapse border border-gray-400 px-6 py-3 text-center'>
                {d.Scored}
              </td>
              <td className='border-collapse border border-gray-400 px-6 py-3 text-center'>
                {d.ToGo}
              </td>
              <td className='border-collapse border border-gray-400 px-6 py-3 text-center'>
                {3 * i || ''}
              </td>
              <td className='border-collapse border border-gray-400 px-6 py-3 text-center'>
                {d.Hits?.join(', ')}
              </td>
            </tr>
          );
        })}
      </tbody>
    </table>
  );
};

export default ScoreBoard;
