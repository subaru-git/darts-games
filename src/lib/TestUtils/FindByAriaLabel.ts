export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
