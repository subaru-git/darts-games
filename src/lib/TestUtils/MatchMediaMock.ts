import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
