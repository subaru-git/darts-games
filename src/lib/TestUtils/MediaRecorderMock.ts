export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
