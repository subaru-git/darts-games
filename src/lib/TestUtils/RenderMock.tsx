import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
