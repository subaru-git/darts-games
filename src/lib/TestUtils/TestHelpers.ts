import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}