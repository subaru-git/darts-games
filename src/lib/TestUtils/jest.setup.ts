import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
