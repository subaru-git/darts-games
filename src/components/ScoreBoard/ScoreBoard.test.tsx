import ScoreBoard from './ScoreBoard';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(<ScoreBoard data={[]} />);
  expect(container).toMatchSnapshot();
});
