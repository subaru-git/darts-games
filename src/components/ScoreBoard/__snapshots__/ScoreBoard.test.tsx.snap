// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <table
    class="table-fixed border-collapse"
  >
    <thead>
      <tr>
        <th
          class="border-collapse border border-gray-400 px-6 py-3"
        >
          Scored
        </th>
        <th
          class="border-collapse border border-gray-400 px-6 py-3"
        >
          To Go
        </th>
        <th
          class="border-collapse border border-gray-400 px-6 py-3"
        >
          Darts
        </th>
        <th
          class="border-collapse border border-gray-400 px-6 py-3"
        >
          Hits
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
